libpod/sqlite_state.go

```diff
@@ -1340,47 +1315,22 @@ func (s *SQLiteState) HasPod(id string) (bool, error) {
 		return false, define.ErrDBClosed
 	}
 
-	return false, define.ErrNotImplemented
-
-	// podID := []byte(id)
-
-	// exists := false
+	row := s.conn.QueryRow("SELECT 1 FROM PodConfig WHERE ID=?;", id)
 
-	// db, err := s.getDBCon()
-	// if err != nil {
-	// 	return false, err
-	// }
-	// defer s.deferredCloseDBCon(db)
-
-	// err = db.View(func(tx *bolt.Tx) error {
-	// 	podBkt, err := getPodBucket(tx)
-	// 	if err != nil {
-	// 		return err
-	// 	}
-
-	// 	podDB := podBkt.Bucket(podID)
-	// 	if podDB != nil {
-	// 		if s.namespaceBytes != nil {
-	// 			podNS := podDB.Get(namespaceKey)
-	// 			if bytes.Equal(s.namespaceBytes, podNS) {
-	// 				exists = true
-	// 			}
-	// 		} else {
-	// 			exists = true
-	// 		}
-	// 	}
-
-	// 	return nil
-	// })
-	// if err != nil {
-	// 	return false, err
-	// }
+	var check int
+	if err := row.Scan(&check); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
```

Comment:
we seem to do this a lot in the DB code (including the container code). do we need something like IsEmpty|NoRows ?

Reply:
Wouldn't save us much in terms of code... Go is super verbose around errors, no fixing that.